sparse/unit_test/Test_Sparse_spmv.hpp

```diff
@@ -1450,8 +1450,7 @@ void test_spmv_bsrmatrix_controls_pattern(
     // for CUDA 9, Kokkos half is actually float. However, the tensor core SpMV
     // uses CUDA's half type, not Kokkos, so we still need a reduced precision
     // test.
-    double eps =
-        2 * KOKKOSKERNELS_IMPL_FP16_EPSILON * KOKKOSKERNELS_IMPL_FP16_RADIX;
+    double eps = std::numeric_limits<Kokkos::Experimental::half_t>::epsilon();
```

Comment:
@cwpearson, @brian-kelley: Shouldn't this epsilon from the `test_y::value_type`...?

Reply:
@e10harvey Yeah, I don't see why this should be hardcoded to the epsilon for half_t. Following up the calls, this test function gets called with all scalar types.

Reply:
@lucbv: Did I miss this during the review of #1412 or is the tolerance correct here?

Reply:
Yes, I would be okay with hlaf_t if the type was clearly hard coded inside the function but here it is implicitly assumed that the control structure will use the tensor core unit and that all tensor core unit are (and always will?) use half precision. This needs to be reworked a bit

Reply:
The tensor cores always compute at half precision on volta